commit 98445362cf6eb6fff625574cd668d70eb8e2493c
Author: Phil Elwell <phil@raspberrypi.org>
Date:   Mon Oct 29 14:45:45 2018 +0000

    Revert "rtc: pcf8523: properly handle oscillator stop bit"
    
    This reverts commit ede44c908d44b166a5b6bd7caacd105c2ff5a70f.
    
    See: https://github.com/raspberrypi/firmware/issues/1065
    
    Signed-off-by: Phil Elwell <phil@raspberrypi.org>

diff --git a/drivers/rtc/rtc-pcf8523.c b/drivers/rtc/rtc-pcf8523.c
index 3fcd2cbafc84..5abfcaf5c937 100644
--- a/drivers/rtc/rtc-pcf8523.c
+++ b/drivers/rtc/rtc-pcf8523.c
@@ -201,8 +201,28 @@ static int pcf8523_rtc_read_time(struct device *dev, struct rtc_time *tm)
 	if (err < 0)
 		return err;
 
-	if (regs[0] & REG_SECONDS_OS)
-		return -EINVAL;
+	if (regs[0] & REG_SECONDS_OS) {
+		/*
+		 * If the oscillator was stopped, try to clear the flag. Upon
+		 * power-up the flag is always set, but if we cannot clear it
+		 * the oscillator isn't running properly for some reason. The
+		 * sensible thing therefore is to return an error, signalling
+		 * that the clock cannot be assumed to be correct.
+		 */
+
+		regs[0] &= ~REG_SECONDS_OS;
+
+		err = pcf8523_write(client, REG_SECONDS, regs[0]);
+		if (err < 0)
+			return err;
+
+		err = pcf8523_read(client, REG_SECONDS, &regs[0]);
+		if (err < 0)
+			return err;
+
+		if (regs[0] & REG_SECONDS_OS)
+			return -EAGAIN;
+	}
 
 	tm->tm_sec = bcd2bin(regs[0] & 0x7f);
 	tm->tm_min = bcd2bin(regs[1] & 0x7f);
@@ -238,7 +258,6 @@ static int pcf8523_rtc_set_time(struct device *dev, struct rtc_time *tm)
 		return err;
 
 	regs[0] = REG_SECONDS;
-	/* This will purposely overwrite REG_SECONDS_OS */
 	regs[1] = bin2bcd(tm->tm_sec);
 	regs[2] = bin2bcd(tm->tm_min);
 	regs[3] = bin2bcd(tm->tm_hour);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1001_98445362cf6eb6fff625574cd668d70eb8e2493c.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1001_98445362cf6eb6fff625574cd668d70eb8e2493c.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
